documentation/secret_docs.md

```diff
@@ -27,10 +29,38 @@ Note that either `slack_access_token` or `slack_hooks` must be defined. If both
 
 Some operations, such as fetching a config file from a private repository, or the commit corresponding to a commit comment event, require a personal access token. Refer [here](https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token) for detailed instructions on token generation.
 
+*See `repos` if you need to support multiple repositories that use different tokens.*
+
 ## `gh_hook_token`
 
 Refer [here](https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/securing-your-webhooks) for more information on securing webhooks with a token.
 
+*See `repos` if you need to support multiple repositories that use different tokens.*
+
+## `repos`
+
+If you're using Monorobot for multiple repositories that need different secrets (e.g., one on github.com and another on GitHub Enterprise), you can provide them as an object. Secrets defined here will take precedence over those defined at the top level of the secrets file.
+
+Repository URLs should be fully qualified (include the protocol), with no trailing backslash.
+
+```json
+{
+  "https://github.com/ahrefs/runner" : {
+    "gh_token": "XXX"
+  },
+  "https://git.ahrefs.com/ahrefs/coyote" : {
+    "gh_token": "XXX",
+    "gh_hook_token": "XXX"
+  }
+}
+```
+
+## `allowed_repos`
+
+Use this option to restrict incoming notifications from GitHub to approved repository URLs.
+
+Repository URLs should be fully qualified (include the protocol), with no trailing backslash.
+
```

Comment:
I don't understand why this second value is required. Aren't all the whitelisted repos already listed under `repos`?

Reply:
This was to 1) stay compatible with existing configs, 2) let repos be whitelisted without defining repo-specifc secrets, and 3) let whitelisting itself be optional. I could alternatively use `repos` as the whitelist (and expect null for whitelisted repos that use default secret), or rename to something less confusing like `repo_secrets`

Reply:
I think we should break compatibility in such cases, or support two formats - legacy and new one (have one object per repository with secrets and any other configurations necessary per-repo)